packages/services/schema/src/lib/federation-tag-extraction.ts

```diff
@@ -472,10 +353,44 @@ export function applyTagFilterOnSubgraphs<
     name: string;
   },
 >(subgraphs: Array<TType>, filter: Federation2SubgraphDocumentNodeByTagsFilter): Array<TType> {
-  const filteredSubgraphs = subgraphs.map(subgraph => ({
-    ...subgraph,
-    ...applyTagFilterToInaccessibleTransformOnSubgraphSchema(subgraph.typeDefs, filter),
-  }));
+  const specImpl = (resolveName: (name: string) => string) => (subgraph: TType) => {
+    const inaccessibleName = resolveName('@inaccessible');
+    const tagName = resolveName('@tag');
+    const filteredSubgraph = {
+      ...subgraph,
+      ...applyTagFilterToInaccessibleTransformOnSubgraphSchema(
+        subgraph.typeDefs,
+        filter,
+        inaccessibleName,
+        tagName,
+      ),
+    };
+    return filteredSubgraph;
+  };
+
+  // for every subgraph, get the spec and apply it.
+  // Current behavior is to treat fed1 and fed2 the same and support all versions.
+  const federationTagFilterSpec = new LinkableSpec('https://specs.apollo.dev/federation', {
+    [FEDERATION_V1]: specImpl,
+    v1_1: specImpl,
+    v2_9: specImpl,
+  });
+
+  let filteredSubgraphs = subgraphs.map(subgraph => {
```

Comment:
I originally intended the LinkableSpec to handle all the implementation for a directive, but since this directive impacts other subgraphs, the logic has to be split.

I admit this makes it a bit more complicated to read and comprehend. There may be a few options to simplify. E.g.:
* expose an API for interacting with multiple subgraphs
* compose first, then apply the spec logic on the one supergraph

Reply:
> compose first, then apply the spec logic on the one supergraph

Transforming `@tag` to `@inaccessible` and passing it to the composition library was the way of doing things here, because I rather trust the federation library giving me a valid supergraph than post-processing the supergraph and potentially generating an invalid supergraph